parlai/mturk/core/react_server/dev/app.jsx

```diff
@@ -235,6 +238,170 @@ class MainApp extends React.Component {
   }
 }
 
-var main_app = <MainApp />;
+
+// TODO consolidate shared functionality from SocketManager in a way that
+// prevents this class from setting a whole lot of dummy methods
+class StaticApp extends React.Component {
+  constructor(props) {
+    super(props);
+    let initialization_status = 'initializing';
+    if (!doesSupportWebsockets()) {
+      initialization_status = 'websockets_failure';
+    }
+
+    // TODO move constants to props rather than state
```

Comment:
this looks like you haven't finalized it

Reply:
I'm not sure I follow. This and many TODOs are due to the fact that this feature is a v0 baseline implementation and I want pointers that let me remember what I have to fix in future work